tests/test_channel_metrics/test_completely_bounded_trace_norm.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+from toqito.channels import dephasing
+def test_diamond_norm_quantum_channel():
+    """The diamond norm of a quantum channel is 1"""
+    phi = dephasing(1)
+    np.testing.assert_equal(np.isclose(diamond_norm(phi), 1, atol=1e-3), True)
+def test_diamond_norm_unitary():
+    """The diamond norm of phi = id- U id U* is the diameter of the smallest circle that contains the eigenvalues of U"""
+    U = 1 / np.sqrt(2) * np.array([[1, 1], [-1, 1]]) # Hadamard gate
+    phi = np.array([[np.eye(2), np.eye(2)], [U, -U]])
+    phi = phi.reshape((4, 4))
+    lam, eigv = np.linalg.eig(U)
+    diameter = np.abs(lam[0] - lam[1])
+    np.testing.assert_equal(np.isclose(diamond_norm(phi), diameter, atol=1e-3), True)
+def test_diamond_norm_non_square():
+    """Non-square inputs for diamond norm."""
+    with np.testing.assert_raises(ValueError):
+        phi = np.array([[1, 2, 3], [4, 5, 6]])
+        diamond_norm(phi)
+
+if __name__ == "__main__":
```

Comment:
It seems odd to me that the `test_completely_bounded_trace_norm` file is testing the diamond norm functionality, right? We probably should also add explicit tests for the `completely_bounded_spectral_norm` as well.

Reply:
I wanted to call `completely_bounded_trace_norm` diamond norm, as in the Wikipedia article, which is easy to confuse with the diamond norm distance functionality in toqito. But sticking to `completely_bounded_trace_norm` is maybe easier in implementation.

Reply:
Ah, I see where the issue here might be:

- Indeed, the completely bounded trace norm is equivalent to the diamond norm (you are completely correct)
- The `diamond_norm` function as is present on `toqito` computes the SDP for the diamond norm. 
- My mistake in writing the title of this issue is that I should have specified that this is for the "completely bounded norm" (not just the "completely bounded trace norm").

There shouldn't be too much to change on your end though as the completely bounded norm of a channel is equal to the diamond norm of the dual map.

Does that make sense? Let me know if anything I've said above isn't clear, and I'll be happy to clarify. Thank you again!

Reply:
According to [sec. 4](http://theoryofcomputing.org/articles/v005a011) the SDP for the distance between two channels is simpler than the SDP for the completely bounded trace norm of channels (sec. 3). Have you implemented the SDP of sec. 4 in  `diamond_norm.py` @vprusso ? 
I have not implemented SDP from sec.3, but the [QETLAB](https://github.com/nathanieljohnston/QETLAB/blob/master/DiamondNorm.m) SDP, which is also cited in Wikipedia. Yet, i have not seen a mathematical derivaton of this formulation.

Reply:
Hi @AmanieOxana 

Right, the `diamond_norm.py` file in `toqito` (should be) implementing the diamond norm as specified in this QETLAB file:
https://github.com/nathanieljohnston/QETLAB/blob/master/DiamondNorm.m

And, ideally, the `completely_bounded_norm.py` file in `toqito` for this feature would implement the same functionality as in this file on QETLAB:
https://github.com/nathanieljohnston/QETLAB/blob/master/CBNorm.m

Does that clear things up or have I made things more confusing?

Reply:
It is a new day and I have a cup of coffee now. I believe (somehow) I missed what you are saying here (which you were saying perfectly clearly before, I just missed it). 

I agree with everything you've written above. Thank you for sticking with me and your patience :)